src/GT-Debugger/GTSpecPreDebugWindow.class.st

```diff
@@ -179,26 +163,24 @@ GTSpecPreDebugWindow >> initializeCodePane [
 
 { #category : #initialization }
 GTSpecPreDebugWindow >> initializePresenter [
-
 	super initializePresenter.
-	
-	debugger whenChangedDo: [ :aDebugger |
-		aDebugger 
-			ifNil: [ self clearWidget ] 
-			ifNotNil:  [ self rebuildWidget ]	 ]
 
+	debugger whenChangedDo: [ :aDebugger | aDebugger ifNotNil: [ self rebuildWidget ] ]
 ]
 
 { #category : #'initialization widgets' }
 GTSpecPreDebugWindow >> initializeStackPane [
 	self stackPane
-		displayBlock: [ :aContext | self columnsFor: aContext ];
+		addColumn: (StringTableColumn title: 'Class' evaluated: [ :aContext | self debugger formatStackClassColumnForContext: aContext ]);
+		addColumn: (StringTableColumn title: 'Method' evaluated: [ :aContext | self debugger formatStackMethodColumnForContext: aContext ]);
+		addColumn: (StringTableColumn title: 'Extra' evaluated: [ :aContext | self debugger formatStackExtraColumnForContext: aContext ]);
 		items: self filteredStack;
 		whenSelectionChangedDo: [ :selection | 
-			[ :aContext | 
-			"Set the selection before, as debugAction removes the link with the debugger. "
-			self debugger stackPresentation selection: aContext.
-			self openFullDebugger ] cull: selection selectedItem ]
+			selection selectedItem
```

Comment:
I would prefer to extract this into a `openFullDebuggerOnContext: selection selectedItem` method

Reply:
Ok with me.

Reply:
So cyril? is it ready for integration?